website/docs: stages cleanup (#21558)
goauthentik/authentik: website/docs/add-secure-apps/flows-stages/stages/prompt/index.md

title: Prompt stage
---

> The Prompt stage displays arbitrary fields to the user and stores the submitted values in flow context.

Reviewer: ```suggestion
The Prompt stage prompts the user to enter information. The submitted values are then stored in the flow context.
```

## Overview

> Use this stage when a flow needs to collect custom input, such as profile attributes, passwords, invitation tokens, file uploads, or approval choices.

Reviewer: ```suggestion
Use this stage to collect user input, such as profile attributes, passwords, invitation tokens, file uploads, or approval choices.
```

- **placeholder**: placeholder text, or for choice fields, the available choices.
- **initial value**: pre-filled value, or for choice fields, the default selected choice.
- **order**: numeric ordering among the prompt fields.
- **Interpret placeholder as expression**: evaluate the placeholder dynamically.
- **Interpret initial value as expression**: evaluate the initial value dynamically.

For **Radio Button Group** and **Dropdown**, the placeholder defines the valid choices. It can be a plain string, a list of values, or a list of objects with `label` and `value`.

Some field types also have special behavior:

- **Username** checks for uniqueness.
- **Password** fields within the same stage must all match.
- **Hidden** and **Static** use their configured values without user editing.
- **Radio Button Group** and **Dropdown** only allow values from the defined choice set.

## Flow integration

Use this stage anywhere a flow needs user-provided input.

Common follow-up stages include:

- [User Write](../user_write/index.md) to persist collected values
- [Email](../email/index.md) or [Invitation](../invitation/index.md) to act on collected data

> - policy checks that read from `request.context["prompt_data"]`

Reviewer: ```suggestion
Common follow-ups include:

- A [User Write](../user_write/index.md) stage to persist collected values
- An [Email](../email/index.md) or [Invitation](../invitation/index.md) stage to act on collected data
- Policy checks that read from `request.context["prompt_data"]`
```
